### ISSUE TITLE
Use HTML formatting when creating a new component

### ISSUE THREAD

**Issue by astrofrog** [glue-viz/glue#574]
When defining new components, the use of ``{}`` to indicate attributes might be improved by instead using HTML highlighting syntax. For example by doing

```
text.setHtml('<font color="red"><b>data.ra</b></font> + <font color="blue"><b>data2.ra</b></font>')
```

we can get:

![screen shot 2015-02-25 at 2 50 44 pm](https://cloud.githubusercontent.com/assets/314716/6371627/f5f12c8c-bcfd-11e4-9979-4c98df1c3ddf.png)

It may even be possible to use CSS and make them look like tags which would improve readability a lot.

This would require writing a small translation layer to/from HTML but it might be worth it for readability.

cc @stargaser @ChrisBeaumont

**Comment by astrofrog** [comment]
(I can hack around with this later to see if it would actually work)

**Comment by astrofrog** [comment]
Turns out it's very easy :)

![screen shot 2015-02-25 at 16 25 19](https://cloud.githubusercontent.com/assets/314716/6373442/028c112a-bd0b-11e4-88cc-1bd5facc0fd6.png)

Now of course this is just a little visual formatting but it might be worth thinking about whether we can make this more readable. For example, maybe dataset names should be sanitized by default to not include extensions. Maybe we can also get rid of the parentheses and instead present it like

**2MASS_k.Pixel_x**

Anyway, @ChrisBeaumont - if you have any thoughts about this, let me know and I'll work on a more comprehensive PR if needed.

**Comment by ChrisBeaumont** [comment]
My only motivation for the brackets was that, if you wanted to, you could
type it out. But maybe readability is more important than typability. I
wonder if it's also possible to add hover text over specific words? Then
you could also abbreviate the labels

On Wednesday, February 25, 2015, Thomas Robitaille <notifications@github.com>
wrote:

> Turns out it's very easy :)
>
> [image: screen shot 2015-02-25 at 16 25 19]
> <https://cloud.githubusercontent.com/assets/314716/6373442/028c112a-bd0b-11e4-88cc-1bd5facc0fd6.png>
>
> Now of course this is just a little visual formatting but it might be
> worth thinking about whether we can make this more readable. For example,
> maybe dataset names should be sanitized by default to not include
> extensions. Maybe we can also get rid of the parentheses and instead
> present it like
>
> *2MASS_k.Pixel_x*
>
> Anyway, @ChrisBeaumont <https://github.com/ChrisBeaumont> - if you have
> any thoughts about this, let me know and I'll work on a more comprehensive
> PR if needed.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/glue-viz/glue/issues/574#issuecomment-75980877>.
>


-- 
*************************************
Chris Beaumont
Senior Software Engineer
Harvard Center for Astrophysics
60 Garden Street, MS 42
Cambridge, MA 02138
chrisbeaumont.org
*************************************

**Comment by astrofrog** [comment]
@ChrisBeaumont - well it's possible that I could get it so that you can do either (and once a pair of ``{}`` is formed, it changes to be nicely formatted). In the above you can still change e.g. the ``2MASS_k`` to ``2MASS_j`` in the blue and it will work. I was also wondering about the hover capability, that could work too.

**Comment by stargaser** [comment]
+1 for the color formatting and for abbreviating the labels.

It would be nicer than all the underscores, especially since the w5_psc votable from the tutorial has so many underscores already.

**Comment by astrofrog** [comment]
@ChrisBeaumont @stargaser - see https://github.com/glue-viz/glue/pull/575 :smile:

**Comment by astrofrog** [comment]
Closed by #575
